### PR TITLE
[Fixed] Reset feature does not clear the date

### PR DESCRIPTION
**Pull request by niravkrish**
### Description (*)

Form Reset feature does not clear the UI date field.
I have added code to clear date field in admin UI Forms.

### Fixed Issues (if relevant)

1. magento/magento2#22940: Reset feature does not clear the date

### Steps to reproduce (*)
Admin > Marketing > Catalog/Cart price rule > Add New Rule > Fill all data with date > now click on reset > Review all the fields are cleared or not

### Manual testing scenarios (*)

1. Manual Testing 
2. I have tested this changes with described steps and it is working fine.

### Expected result (*)
All fields should be blank.

### Actual result (*)
date-reset-issue

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by VladimirZaets — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets, thank you for the review.
[ENGCOM-5155](https://jira.corp.magento.com/browse/ENGCOM-5155) has been created to process this Pull Request

**Comment by VasylShvorak** [comment]
With PR changes date is always cleared by Reset button.
**Manual testing scenario:**

- Create Cart price rule with date (fields From and To)
- Go to Cart price rule edit page
- Edit Rule Name and Description, but don't touch date
- Click Reset

**Result:**
:heavy_check_mark: Rule Name and Description are reset to previous values
:x: Date fields From and To are cleared
**Notice**: Dates are cleared only visual. If save Cart price rule after Reset button was clicked, dates values are same as before edit
@VladimirZaets Could you you check if it is expected behavior?

File: app/code/Magento/Ui/view/base/web/js/form/form.js [hunk 1]
```diff
@@ -339,6 +339,7 @@ define([
          */
         reset: function () {
             this.source.trigger('data.reset');
+            $('._has-datepicker').val('');
```
Comment (orlangur) [inline app/code/Magento/Ui/view/base/web/js/form/form.js]: Business logic must not rely on CSS classes, see [Code Demarcation Standard](https://devdocs.magento.com/guides/v2.3/coding-standards/code-standard-demarcation.html). Please add appropriate data attributes instead.

**Review by orlangur — CHANGES_REQUESTED** [review]

File: app/code/Magento/Ui/view/base/web/js/form/form.js (hunk 1 above)
Comment (p-bystritsky) [reply app/code/Magento/Ui/view/base/web/js/form/form.js]: @orlangur updated.

**Comment by p-bystritsky** [comment]
@orlangur please review changes.

File: app/code/Magento/Ui/view/base/web/js/form/form.js [hunk 2]
```diff
@@ -339,6 +339,7 @@ define([
          */
         reset: function () {
             this.source.trigger('data.reset');
+            $('[data-index$="_date"] input').val('');
```
Comment (orlangur) [inline app/code/Magento/Ui/view/base/web/js/form/form.js]: We must not rely on `input` tag name. Please omit unnecessary quotes `""`.

**Review by orlangur — CHANGES_REQUESTED** [review]

File: app/code/Magento/Ui/view/base/web/js/form/form.js (hunk 2 above)
Comment (p-bystritsky) [reply app/code/Magento/Ui/view/base/web/js/form/form.js]: @orlangur updated.

**Review by orlangur — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @orlangur, thank you for the review.
[ENGCOM-5155](https://jira.corp.magento.com/browse/ENGCOM-5155) has been created to process this Pull Request

**Comment by VasylShvorak** [comment]
:heavy_check_mark: QA passed

**Pull request merged**